server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -347,10 +346,9 @@ public ClusterState execute(ClusterState currentState) {
             if (found == false) {
                 repositoriesMetadata.add(new RepositoryMetadata(request.name(), request.type(), request.settings()));
             }
-            repositories = new RepositoriesMetadata(repositoriesMetadata);
-            mdBuilder.putDefaultProjectCustom(RepositoriesMetadata.TYPE, repositories);
             changed = true;
-            return ClusterState.builder(currentState).metadata(mdBuilder).build();
+            var updatedRepositories = new RepositoriesMetadata(repositoriesMetadata);
```

Comment:
Can we keep this line unchanged, i.e. no new name and having it before `changed = true`? I have similar suggestions for other places too. In general, I prefer we keep the code as is as much as possible outside the targeted refactoring.

Reply:
I had to extract the variable to be able to pass it to the lambda inside `currentState.copyAndUpdateProject` - because `repositories` is otherwise a reassigned variable. I can still move the `updatedRepositories` line to before the `changed = true` if that's desired. Out of curiosity, is there a reason you'd like to have that specific format?

Reply:
I don't particularly like the format. My intention was to avoid code change that is _not_ absolutely necessary for specific refactoring. This is my general preference to keep each PR narrow focused. In this case, can we inline the variable since the existing code also inline similar variable in a few other places?

Reply:
Can you explain why you don't like the format? I personally do feel like it's less verbose than the original/your suggestion. Also, if "we" don't like the format, I'd argue the `copyAndUpdateProject` method shouldn't exist. I didn't create that method, so I assume at least one other person preferred that format too.

I think the code changes that are not absolutely necessary for this refactoring are very minimal and acceptable.

Reply:
> Can you explain why you don't like the format? 

By "I don't particularly like the format", I mean that I am not attached to the existing code format, _not_ the new format you are proposing.

> I think the code changes that are not absolutely necessary for this refactoring are very minimal and acceptable.

I think this is the main difference that we have. I'd prefer "code changes that are not absolutely necessary for this refactoring" to be zero, or very close to zero if possible. This is the main reason for my suggestion of not using `copyAndUpdateProject`, not that I don't like the method in general. Just not in this PR. I hope this clarifies it.